### ISSUE TITLE
[NEW FEATURE] how to close the session manually like session.close()

### ISSUE THREAD

**Issue by xishandong** [rawandahmad698/noble-tls#14]
When I use noble_tls, I need to frequently create new sessions to request URLs, but I find that after a while, the system's memory usage becomes quite high. It turns out that the connections are occupying a large amount of memory. When I shut down and restart the application service (as opposed to ending the Python process), the memory is released. Here is my usage demo. Is there a method to support manually closing connections to release session objects?

**Comment by xishandong** [comment]
@app.post("/v4/universal/get")
async def universal_get(body: UniversalModel):
    url = body.url
    universal_headers = json.loads(body.headers)
    universal_proxy = body.proxy
    global_session = noble_tls.Session(client=identifier(), random_tls_extension_order=True)
    response = await global_session.get(url, headers=universal_headers, proxy=universal_proxy)
    return Response(content=response.text, status_code=response.status_code)

**Comment by xishandong** [comment]
def close(self) -> str:
        destroy_session_payload = {
            "sessionId": self._session_id
        }

        destroy_session_response = destroySession(dumps(destroy_session_payload).encode('utf-8'))
        # we dereference the pointer to a byte array
        destroy_session_response_bytes = ctypes.string_at(destroy_session_response)
        # convert our byte array to a string (tls client returns json)
        destroy_session_response_string = destroy_session_response_bytes.decode('utf-8')
        # convert response string to json
        destroy_session_response_object = loads(destroy_session_response_string)

        free_memory(destroy_session_response_object['id'].encode('utf-8'))

        return destroy_session_response_string

**Comment by xishandong** [comment]
Here is the close method I mimicked from tls-client. Will this method also effectively release memory in an asynchronous context?

**Comment by xishandong** [comment]
Based on my testing, this approach doesn't seem to have a significant impact. Do you have any recommended methods or similar implementations that could support memory release needs?

**Comment by ahmedmani** [comment]
im wondering about this as well, is there a reliable method to close the session?

**Comment by larry-ct** [comment]
![image](https://github.com/user-attachments/assets/718e73ed-8ed4-499c-9445-30bb95a2e577)
默认 是会释放内存的

**Comment by xishandong** [comment]
> ![image](https://github.com/user-attachments/assets/718e73ed-8ed4-499c-9445-30bb95a2e577) 默认 是会释放内存的

Yes, but after about 5 hours, the memory usage goes up to 90%. After restarting the script, the memory usage returns to normal.So I'm considering whether there are better options for releasing memory.
